src/type_methods.jl

```diff
@@ -0,0 +1,610 @@
+# defines common methods for the various rotation types
```

Comment:
In general, I find this file a bit scary and it's hard to see what features I get from it (I realize these functions are used elsewhere, bit this seems like *a lot* of effort).